### ISSUE TITLE
T cannot be JSON unwrapped

### ISSUE THREAD

**Issue by scoquelin** [stoicflame/enunciate#141]
Hi,

I am using latest enunciate-maven-plugin version 2.0.0-M.4
I get this error (**getContent(): T cannot be JSON unwrapped**) when this Spring class gets inspected :

https://github.com/spring-projects/spring-hateoas/blob/master/src/main/java/org/springframework/hateoas/Resource.java

Exception is thrown here :

https://github.com/stoicflame/enunciate/blob/v2/jackson/src/main/java/com/webcohesion/enunciate/modules/jackson/model/TypeDefinition.java#L188

where typeMirror is an instance of DecoratedTypeVariable hence why Exception is thrown.

Did I miss some configuration option to avoid this ?
Here is the stack trace,
Thanks

````
[DEBUG] [ENUNCIATE] Added org.springframework.hateoas.PagedResources as a Jackson type definition.
[DEBUG] [ENUNCIATE] Added org.springframework.hateoas.PagedResources.PageMetadata as a Jackson type definition.
[DEBUG] [ENUNCIATE] Added org.springframework.hateoas.Resources as a Jackson type definition.

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 28.718 s
[INFO] Finished at: 2015-09-14T15:41:22-04:00
[INFO] Final Memory: 390M/779M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.4:assemble (assemble) on project <artifactId>: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.4:assemble failed: com.webcohesion.enunciate.EnunciateException: getContent(): T cannot be JSON unwrapped. OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.4:assemble (assemble) on project <artifactId>: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.4:assemble failed: com.webcohesion.enunciate.EnunciateException: getContent(): T cannot be JSON unwrapped.
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:355)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:155)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:216)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:160)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution assemble of goal com.webcohesion.enunciate:enunciate-maven-plugin:2.0.0-M.4:assemble failed: com.webcohesion.enunciate.EnunciateException: getContent(): T cannot be JSON unwrapped.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:143)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	... 19 more
Caused by: java.lang.RuntimeException: com.webcohesion.enunciate.EnunciateException: getContent(): T cannot be JSON unwrapped.
	at com.sun.tools.javac.main.Main.compile(Main.java:553)
	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
	at com.webcohesion.enunciate.Enunciate.run(Enunciate.java:650)
	at com.webcohesion.enunciate.mojo.ConfigMojo.execute(ConfigMojo.java:352)
	at com.webcohesion.enunciate.mojo.AssembleBaseMojo.execute(AssembleBaseMojo.java:86)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
	... 20 more
Caused by: com.webcohesion.enunciate.EnunciateException: getContent(): T cannot be JSON unwrapped.
	at com.webcohesion.enunciate.modules.jackson.model.TypeDefinition.aggregatePotentialAccessors(TypeDefinition.java:188)
	at com.webcohesion.enunciate.modules.jackson.model.TypeDefinition.loadPotentialAccessors(TypeDefinition.java:143)
	at com.webcohesion.enunciate.modules.jackson.model.TypeDefinition.<init>(TypeDefinition.java:87)
	at com.webcohesion.enunciate.modules.jackson.model.ObjectTypeDefinition.<init>(ObjectTypeDefinition.java:35)
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.createTypeDefinition(EnunciateJacksonContext.java:215)
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:429)
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext$ReferencedJsonDefinitionVisitor.visitDeclared(EnunciateJacksonContext.java:396)
	at com.webcohesion.enunciate.javac.decorations.type.DecoratedDeclaredType.accept(DecoratedDeclaredType.java:83)
	at com.webcohesion.enunciate.modules.jackson.EnunciateJacksonContext.addReferencedTypeDefinitions(EnunciateJacksonContext.java:371)
	at com.webcohesion.enunciate.modules.jackson.JacksonModule.addDataTypeDefinitions(JacksonModule.java:107)
	at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.addReferencedDataTypeDefinitions(JaxrsModule.java:149)
	at com.webcohesion.enunciate.modules.jaxrs.JaxrsModule.call(JaxrsModule.java:86)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:31)
	at com.webcohesion.enunciate.io.InvokeEnunciateModule.onNext(InvokeEnunciateModule.java:10)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:79)
	at rx.internal.operators.OperatorSingle$ParentSubscriber.onCompleted(OperatorSingle.java:109)
	at rx.internal.operators.NotificationLite.accept(NotificationLite.java:140)
	at rx.internal.operators.TakeLastQueueProducer.emit(TakeLastQueueProducer.java:100)
	at rx.internal.operators.TakeLastQueueProducer.startEmitting(TakeLastQueueProducer.java:45)
	at rx.internal.operators.OperatorTakeLast$1.onCompleted(OperatorTakeLast.java:59)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:658)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.drainAndComplete(OperatorMerge.java:521)
	at rx.internal.operators.OperatorMerge$MergeSubscriber.completeInner(OperatorMerge.java:486)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.emit(OperatorMerge.java:673)
	at rx.internal.operators.OperatorMerge$InnerSubscriber.onCompleted(OperatorMerge.java:600)
	at rx.internal.util.ScalarSynchronousObservable$ScalarSynchronousAction.call(ScalarSynchronousObservable.java:117)
	at rx.schedulers.ExecutorScheduler$ExecutorAction.run(ExecutorScheduler.java:173)
	at rx.schedulers.ExecutorScheduler$ExecutorSchedulerWorker.run(ExecutorScheduler.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: com.webcohesion.enunciate.EnunciateContext.class
	at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:101)
	at rx.internal.operators.OperatorDoOnEach$1.onNext(OperatorDoOnEach.java:81)
	... 23 more
````

**Comment by stoicflame** [comment]
It's a bug. Thanks for the report.

Did you want that resource included in your API? I think Enunciate might be a bit too aggressive in it's inclusion policy, so we'll draw it back a bit. This might help #137, too...

**Comment by stoicflame** [comment]
Fixed at c83588d.

Also, at a94e144 and 1b01e1b, the datatype detection strategy was narrowed so that class won't show up in your API unless you explicitly include it in the config.

**Comment by scoquelin** [comment]
Thanks a lot @stoicflame !
